clang/test/SemaSYCL/spurious-host-warning-ReqdSubGroupSize.cpp

```diff
@@ -0,0 +1,14 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-host -fsyntax-only -verify %s -DSYCLHOST
```

Comment:
Do we have a better existing place for this?  Reqd-work-group-size.cpp already has a host test.

Reply:
Done

Reply:
> Do we have a better existing place for this? Reqd-work-group-size.cpp already has a host test.

No - it doesn't.

Reply:
> 
> 
> > Do we have a better existing place for this? Reqd-work-group-size.cpp already has a host test.
> 
> No - it doesn't.

Well, yes, it DOES, it just so happens to be Reqd-WORK-group-size.cpp, NOT SUB-group size :)

Reply:
Sorry, missed that, but in both cases 0 lines of test code is shared between device and host compilation checks, so I think combining these checks into single test source makes it difficult to maintain.
Instead of:
```c++
// run device compilation
// run host compilation
#ifdef __SYCL_DEVICE_ONLY__
// device compilation checks
#else
// host compilation checks
#endif
```
I would split them into `reqd-work-group-size[-host|-device].cpp` and `reqd-sub-group-size[-host|-device].cpp`. This seems to be easier to read/maintain.
Does it make sense?

Reply:
> Sorry, missed that

Yep, me too, thats what I was pointing out :)  I must have clicked the wrong test when I was looking to see if the test could be combined.  


> , but in both cases 0 lines of test code is shared between device and host compilation checks, so I think combining these checks into single test source makes it difficult to maintain.
> I would split them into `reqd-work-group-size[-host|-device].cpp` and `reqd-sub-group-size[-host|-device].cpp`. This seems to be easier to read/maintain.
> Does it make sense?

Yep, it does.

Reply:
> I would split them into `reqd-work-group-size[-host|-device].cpp` and `reqd-sub-group-size[-host|-device].cpp`.

@smanna12, would you mind doing this change?

Reply:
@bader I will make this changes and submit updated patches.

Reply:
Thanks!